generator/internal/mixin_utils.cc

```diff
@@ -49,61 +49,80 @@ std::unordered_map<std::string, std::string> const& GetMixinProtoPathMap() {
+      http_rule.set_post(rule_value);
+    } else if (rule_key == "put") {
+      http_rule.set_put(rule_value);
+    } else if (rule_key == "patch") {
+      http_rule.set_patch(rule_value);
+    } else if (rule_key == "delete") {
+      http_rule.set_delete_(rule_value);
+    }
+  }
 }
 
 /**
  * Extract Mixin methods from the YAML file, together with the overwritten http
  * info.
  */
-std::unordered_map<std::string, MixinMethodOverride> GetMixinMethodOverrides(
+std::unordered_map<std::string, google::api::HttpRule> GetMixinHttpOverrides(
     YAML::Node const& service_config) {
-  std::unordered_map<std::string, MixinMethodOverride> mixin_method_overrides;
-  if (service_config.Type() != YAML::NodeType::Map)
-    return mixin_method_overrides;
-  if (!service_config["http"]) return mixin_method_overrides;
+  std::unordered_map<std::string, google::api::HttpRule> http_rules;
+
+  if (service_config.Type() != YAML::NodeType::Map) return http_rules;
+  if (!service_config["http"]) return http_rules;
+
   auto const& http = service_config["http"];
-  if (http.Type() != YAML::NodeType::Map) return mixin_method_overrides;
+  if (http.Type() != YAML::NodeType::Map) return http_rules;
+
   auto const& rules = http["rules"];
-  if (rules.Type() != YAML::NodeType::Sequence) return mixin_method_overrides;
+  if (rules.Type() != YAML::NodeType::Sequence) return http_rules;
+
   for (auto const& rule : rules) {
     if (rule.Type() != YAML::NodeType::Map) continue;
 
     auto const& selector = rule["selector"];
     if (selector.Type() != YAML::NodeType::Scalar) continue;
-    std::string const method_full_name = selector.as<std::string>();
 
-    for (auto const& kv : rule) {
-      if (kv.first.Type() != YAML::NodeType::Scalar ||
-          kv.second.Type() != YAML::NodeType::Scalar)
-        continue;
-
-      std::string const http_verb_lower =
-          absl::AsciiStrToLower(kv.first.as<std::string>());
-      auto const& http_verbs = GetHttpVerbs();
-      auto const it = http_verbs.find(http_verb_lower);
-      if (it == http_verbs.end()) continue;
-      std::string const& http_verb = it->second;
-      std::string const http_path = kv.second.as<std::string>();
-      absl::optional<std::string> http_body;
-      if (rule["body"]) {
-        http_body = rule["body"].as<std::string>();
+    std::string const method_full_name = selector.as<std::string>();
+    google::api::HttpRule http_rule;
+    ParseHttpRule(method_full_name, rule, http_rule);
+
+    if (rule["additional_bindings"]) {
+      auto const& additional_bindings = rule["additional_bindings"];
+      if (additional_bindings.Type() == YAML::NodeType::Sequence) {
+        for (auto const& additional_binding : additional_bindings) {
+          google::api::HttpRule& additional_http_rule =
+              *http_rule.add_additional_bindings();
+          if (additional_binding.Type() != YAML::NodeType::Map) continue;
+          ParseHttpRule(method_full_name, additional_binding,
```

Comment:
It probably doesn't matter, but I don't think the `additional_bindings` should get the parent's selector.

Reply:
That may be my misunderstanding, I thought first `post` is the sibling of the other 3 `post`s.

If not, I should remove the selectors for additional_bindings.

```
- selector: google.iam.v1.IAMPolicy.SetIamPolicy
    post: '/v1/{resource=projects/*/locations/*/connectionProfiles/*}:setIamPolicy'
    body: '*'
    additional_bindings:
    - post: '/v1/{resource=projects/*/locations/*/migrationJobs/*}:setIamPolicy'
      body: '*'
    - post: '/v1/{resource=projects/*/locations/*/conversionWorkspaces/*}:setIamPolicy'
      body: '*'
    - post: '/v1/{resource=projects/*/locations/*/privateConnections/*}:setIamPolicy'
      body: '*'
```

Reply:
Yeah, I double checked, `selector` is empty for a `http_rule` parsed by `MethodDescriptor`. the [selector](https://github.com/googleapis/googleapis/blob/7f9941f4ba22d6eb3bb7fa31f80aae3a1b3b957e/google/api/http.proto#L313) could be a different thing from the selector in YAML.

Reply:
I think the point is that they want to use one proto for both the `foo.proto` annotations and for mixin overrides in the `foo_gapic.yaml`.

The selector field only means something in the `foo_gapic.yaml` so we wont see it set in any of the googleapis protos.

It's probably totally fine to ignore the selector field when turning the yaml -> HttpRule, but my instincts are to parse the HttpRule faithfully, including the selector field, even if that field is ignored in other layers of the generator.